core/src/avm2/object/xml_object.rs

```diff
@@ -212,6 +212,23 @@ impl<'gc> TObject<'gc> for XmlObject<'gc> {
             }
         }
 
+        // Special case to handle code like: xml["@attr"]
+        let multiname = if !name.has_explicit_namespace()
+            && !name.is_attribute()
+            && !name.is_any_name()
+            && !name.is_any_namespace()
+        {
+            name.local_name()
+                .map(|name| string_to_multiname(activation, name))
+        } else {
+            None
+        };
+
+        let name = match multiname {
```

Comment:
Can this just be `let name = multiname.or(name)`

Reply:
`or` takes an `Option`, but that made me realize that `as_ref.unwrap_or` works.